docs/operations/snapshot_restore_api.md

```diff
@@ -0,0 +1,94 @@
+# Snapshot Restore API
+
+The `snapshot.kubevirt.io` API Group defines resources for snapshotting and restoring KubeVirt `VirtualMachines`
+
+## Prerequesites
+
+### VolumeSnapshotClass
+
+KubeVirt leverages the `VolumeSnapshot` functionality of Kubernetes [CSI drivers](https://kubernetes-csi.github.io/docs/drivers.html) for capturing persistent `VirtualMachine` state.  So, you should make sure that your `VirtuallMachine` uses `DataVolumes` or `PersistentVolumeClaims` backed by a `StorageClass` that supports `VolumeSnapshots` and a `VolumeSnapshotClass` is properly configured for that `StorageClass`.
+
+To list `VolumeSnapshotClasses`:
+
+```bash
+kubectl get volumesnapshotclass
+```
+
+Make sure that the `provisioner` property of your `StorageClass` matches the `driver` property of the `VolumeSnapshotClass`
+
+Even if you have no `VolumeSnapshotClasses` in your cluster, `VirtualMachineSnapshots` are not totally useless.  They will still backup your `VirtualMachine` configuration.
+
+### Snapshot Feature Gate
+
+Snapshot/Restore are currently considered alpha features and are disabled by default.
+
+```bash
+kubectl patch -n kubevirt kubevirt kubevirt -p '{"spec": {"configuration": { "developerConfiguration": { "featureGates": [ "Snapshot" ] }}}}' -o json --type merge
+```
+
+## Snapshot a VirtualMachine
+
+Snapshot a virtualMachine is supported for online and offline vms.
+
+When snapshoting a running vm the controller will check for qemu guest agent in the vm, if it exists it will freeze the vm filesystems before taking the snapshot (and unfreeze after the snapshot). It is recommended to take online snapshot with the guest agent for a better snapshot, if not present a best effort snapshot will be taken.\
```

Comment:
This comma creates a run-on sentence.

The parenthesis generally means `ignore this stuff` yet the unfreeze process is important here.  Let's remove them for clarity.

Fixed a couple spelling mistakes along the way as well. :)

```
When snapshotting a running vm the controller will check for qemu guest agent in the vm. If the agent exists it will freeze the vm filesystems before taking the snapshot and unfreeze after the snapshot. It is recommended to take online snapshot with the guest agent for a better snapshot, if not present a best effort snapshot will be taken.
```